src/realm/object-store/sync/sync_user.cpp

```diff
@@ -407,7 +413,6 @@ bool SyncUser::has_device_id() const
 
 SyncUser::State SyncUser::state() const
```

Comment:
Conceptually, m_state does need to be protected by a mutex because there are sections of code in SyncUser that rely on m_state not changing after it is read (eg. `SyncUser::session_for_on_disk_path`). But here in `SyncUser::state()` it is rather pointless to use a lock to just read the state and return. Could we keep `m_state` guarded by `m_mutex` but have this method not lock the mutex and exclude the safety check by using `NO_THREAD_SAFETY_ANALYSIS`?

Reply:
Good idea!